File: apps/starter-android/app/src/main/kotlin/com/onlystack/starterapp/core/user/UserModels.kt
package com.onlystack.starterapp.core.user

data class UserPreferencesSummary(
    val theme: String? = null,
    val pushMarketingEnabled: Boolean? = null,
    val pushActivityEnabled: Boolean? = null,
    val pushTransactionalEnabled: Boolean? = null,
    val pushSystemEnabled: Boolean? = null,
    val emailNotificationsEnabled: Boolean? = null,
)

data class UserProfile(
    val id: String,
    val email: String,
    val displayName: String?,
    val preferences: UserPreferencesSummary? = null,
)

data class UpdateUserProfileBody(
    val displayName: String? = null,
)

File: apps/starter-ios/StarterApp/Core/User/UserModels.swift
import Foundation

struct UserPreferencesSummary: Codable, Equatable {
    let theme: String?
    let pushMarketingEnabled: Bool?
    let pushActivityEnabled: Bool?
    let pushTransactionalEnabled: Bool?
    let pushSystemEnabled: Bool?
    let emailNotificationsEnabled: Bool?
}

struct UserProfile: Codable, Equatable {
    let id: String
    let email: String
    let displayName: String?
    let preferences: UserPreferencesSummary?
}

struct UpdateUserProfileBody: Encodable {
    let displayName: String?
}
